From eb3a484c072f7cc5a319d919d54e6d843cd8c79f Mon Sep 17 00:00:00 2001
From: Andreas Sturmlechner <andreas.sturmlechner@gmail.com>
Date: Sun, 7 Oct 2018 12:05:45 +0200
Subject: [PATCH 1/2] Make Qt5Test optional

Test Plan: Builds fine w/o Qt5Test present.

Subscribers: kdevelop-devel

Tags: #kdevelop

Differential Revision: https://phabricator.kde.org/D16012
---
 CMakeLists.txt | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c09ca29..c1e25d6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -21,7 +21,10 @@ include(GenerateExportHeader)
 include(FeatureSummary)
 
 set(QT_MIN_VERSION "5.5.0")
-find_package(Qt5 ${QT_MIN_VERSION} CONFIG REQUIRED Core Widgets Test)
+find_package(Qt5 ${QT_MIN_VERSION} CONFIG REQUIRED Core Widgets)
+if(BUILD_TESTING)
+  find_package(Qt5Test ${QT_MIN_VERSION} CONFIG REQUIRED)
+endif()
 set(KF5_DEP_VERSION "5.15.0")
 find_package(KF5 ${KF5_DEP_VERSION} REQUIRED COMPONENTS Archive ThreadWeaver TextEditor I18n ItemModels KCMUtils)
 find_package(KDevPlatform ${KDEVPLATFORM_VERSION} REQUIRED)
-- 
2.19.1


From 600c167d742da9f22f6d72f6b5d8c3919de2580b Mon Sep 17 00:00:00 2001
From: Andreas Sturmlechner <andreas.sturmlechner@gmail.com>
Date: Wed, 10 Oct 2018 01:35:19 +0200
Subject: [PATCH 2/2] Drop unused KF5Archive

---
 CMakeLists.txt                  | 2 +-
 completion/tests/CMakeLists.txt | 4 ----
 2 files changed, 1 insertion(+), 5 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c1e25d6..5f1070a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -26,7 +26,7 @@ if(BUILD_TESTING)
   find_package(Qt5Test ${QT_MIN_VERSION} CONFIG REQUIRED)
 endif()
 set(KF5_DEP_VERSION "5.15.0")
-find_package(KF5 ${KF5_DEP_VERSION} REQUIRED COMPONENTS Archive ThreadWeaver TextEditor I18n ItemModels KCMUtils)
+find_package(KF5 ${KF5_DEP_VERSION} REQUIRED COMPONENTS ThreadWeaver TextEditor I18n ItemModels KCMUtils)
 find_package(KDevPlatform ${KDEVPLATFORM_VERSION} REQUIRED)
 find_package(KDevelop-PG-Qt REQUIRED)
 
diff --git a/completion/tests/CMakeLists.txt b/completion/tests/CMakeLists.txt
index dbe37c1..214cc05 100644
--- a/completion/tests/CMakeLists.txt
+++ b/completion/tests/CMakeLists.txt
@@ -23,9 +23,5 @@ ecm_add_test(
         Qt5::Test
         KDev::Tests
         KDev::Language
-
-        # TODO: KF5: KDevice is forcing us to have this dependency. I'm sure
-        # we can work-around it :)
-        KF5::Archive
 )
 
-- 
2.19.1

